src/main/components/LoginForm.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+
+import Button from 'material-ui/Button';
+import TextField from 'material-ui/TextField';
+import { FormControlLabel } from 'material-ui/Form';
+import Checkbox from 'material-ui/Checkbox';
+
+/**
+ * The UI Component representing the drop down menu for logging in/signing up.
+ *
+ * @class
+ */
+export class LoginForm extends React.Component {
+    constructor() {
+        super();
+        this.handleClick = this.handleClick.bind(this);
+        this.state = {isSigningUp: false};
+    }
+
+    /**
+     * Toggles the  menu fields for logging in/signing up upon a click event.
+     *
+     * @function
+     */
+    handleClick() {
+        let state = this.state.isSigningUp;
+        state = !state;
+        this.setState({
+            isSigningUp: state
+        });
+    }
+
+    /**
+     * Renders the drop down menu for logging in/signing up.
+     *
+     * @function
+     */
+    render() {
+        let confirmPass, userEmail, userAge, flexSetting, buttonText;
+        if(this.state.isSigningUp) {
```

Comment:
The render method here is pretty hard to read; I'd suggest taking some of these flexible items and moving them into private functions defined above the class scope so that our `render` method is clean and can be easily extended in the future.

```
const confirmPass = isSigningUp => isSigningUp
    ? (<div className="bp--login-field">
        <TextField
            label="Confirm Password"
            type="password"
        />
    </div>)
    : null;

const userEmail = isSigningUp
    ? (<div className="bp--login-field">
        <TextField
            label="Email"
        />
    </div>)
    : null;

const userAge = isSigningUp => isSigningUp
    ? (<div className="bp--login-field">
        <FormControlLabel
            control={
                <Checkbox
                    value="checkedG"
                />
            }
            label="I'm 21 or older."
        />
    </div>)
    : null;

const flexSetting = isSigningUp
    ? (<div className="bp--login-field">
        {"Already have an account?"}
        <a onClick={this.handleClick} href={"#"} className="new_account">
            {"Log In"}
        </a>
    </div>)
    : (<div className="bp--login-field">
        {"Need an account?"}
        <a onClick={this.handleClick} href={"#"} className="new_account">
            {"Sign Up"}
        </a>
    </div>);

export class LoginForm extends React.Component {
    render() {
        const { isSigningUp } = this.state;
        return (
            <div className="login_form">
                <div className="bp--login-field">
                    <TextField
                        label="Username"
                    />
                </div>
                {userEmail(isSigningUp)}
                <div className="bp--login-field">
                    <TextField
                        label="Password"
                        type="password"
                    />
                </div>
                {confirmPass(isSigningUp)}
                <br/>
                {userAge(isSigningUp)}
                <div className="bp--login-field">
                    <Button raised={true} disableFocusRipple={true} disableRipple={true}>{isSigningUp ? "Join" : "Log in"}</Button>
                </div>
                <br/>
                <div className="bp--login-field">
                    <a href={"#"} className="forgot_pass">
                        {"Forgot Password?"}
                    </a>
                </div>
                <br/>
                {flexSetting(isSigningUp)}
            </div>
        );
    }
}
```